src/unitxt/artifact.py

```diff
@@ -412,6 +412,18 @@ def get_raw(obj):
     return shallow_copy(obj)
 
 
+class ArtifactLink(Artifact):
+    new_artifact: Artifact
+    is_self_deprecated: bool
```

Comment:
In think deprecation and linking are distinct properties.

We can deprecate something without having a new link to it.  So every artifact should have a deprecated field.